nomad/job_endpoint.go

```diff
@@ -1023,12 +1023,12 @@ func (j *Job) Plan(args *structs.JobPlanRequest, reply *structs.JobPlanResponse)
 	if aclObj, err := j.srv.ResolveToken(args.AuthToken); err != nil {
 		return err
 	} else if aclObj != nil {
-		if !aclObj.AllowNsOp(structs.DefaultNamespace, acl.NamespaceCapabilitySubmitJob) {
+		if !aclObj.AllowNsOp(args.RequestNamespace(), acl.NamespaceCapabilitySubmitJob) {
```

Comment:
You should confirm that args.RequestNamespace gets set to default and not empty string when nothing is specified. PErhaps there is a test for this already, if not we should write one.

Reply:
Looks like https://github.com/hashicorp/nomad/blob/08817ff27c6fda89fce45df6bb162c4bcace26d0/nomad/structs/structs.go#L210-L215  does this.